### PR TITLE
[local_auth] Fix activity leak in LocalAuthPlugin

### PR DESCRIPTION
**Pull request by 0xZOne**
Fixes flutter/flutter#92412

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Note that unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I [updated pubspec.yaml](https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates) with an appropriate new version according to the [pub versioning philosophy].
- [ ] I updated CHANGELOG.md to add a description of the change.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[the auto-formatter]: https://github.com/flutter/plugins/blob/master/script/tool/README.md#format-code

### CODE REVIEW

**Comment by stuartmorgan** [comment]
@blasten Ping on this review

**Review by blasten — APPROVED** [review]

File: packages/local_auth/android/src/main/java/io/flutter/plugins/localauth/LocalAuthPlugin.java [hunk 1]
```diff
@@ -350,5 +352,11 @@ public void onReattachedToActivityForConfigChanges(ActivityPluginBinding binding
   public void onDetachedFromActivity() {
     lifecycle = null;
     channel.setMethodCallHandler(null);
+    activity = null;
+  }
+
+  @VisibleForTesting
+  /*package*/ final Activity getActivity() {
```
Comment (stuartmorgan) [inline packages/local_auth/android/src/main/java/io/flutter/plugins/localauth/LocalAuthPlugin.java:359]: Please don't check in commented-out code.

**Review by stuartmorgan — APPROVED** [review]
One small nit; once that's fixed I can add the second approval and land this.

**Review by stuartmorgan — CHANGES_REQUESTED** [review]
(Sorry, wrong option in the review UI; that wasn't meant to be an approval yet)

File: packages/local_auth/android/src/main/java/io/flutter/plugins/localauth/LocalAuthPlugin.java (hunk 1 above)
Comment (0xZOne) [reply packages/local_auth/android/src/main/java/io/flutter/plugins/localauth/LocalAuthPlugin.java:359]: > Please don't check in commented-out code.

Thank you for replying.

This note was added intentionally. Package level protection is the default, but adding the note there will be more clear.

Comment (stuartmorgan) [reply packages/local_auth/android/src/main/java/io/flutter/plugins/localauth/LocalAuthPlugin.java:359]: > > Please don't check in commented-out code.
> 
> This note was added intentionally.

I understand that you did it intentionally, but we don't check in things like this.

> Package level protection is the default, but adding the note there will be more clear.

We don't comment standard language behavior.

Comment (0xZOne) [reply packages/local_auth/android/src/main/java/io/flutter/plugins/localauth/LocalAuthPlugin.java:359]: > > > Please don't check in commented-out code.
> > 
> > 
> > This note was added intentionally.
> 
> I understand that you did it intentionally, but we don't check in things like this.
> 
> > Package level protection is the default, but adding the note there will be more clear.
> 
> We don't comment standard language behavior.

Thanks, I got it now.

**Review by stuartmorgan — APPROVED** [review]
Thanks, LGTM

**Pull request merged**